lib/bolt_spec/plans/mock_executor.rb

```diff
@@ -274,12 +274,7 @@ def create_future(scope: nil, name: nil)
           local = Puppet::Parser::Scope::LocalScope.new
 
           # Compress the current scopes into a single vars hash to add to the new scope
-          current_scope = scope.effective_symtable(true)
```

Comment:
It almost looked like this would recursively pass through parent plan scope, but i did not actually see that in my (limited) testing. The main motivation is to standardize across the mock executor (which we are modeling after in our orch executor) and the fiber executor https://github.com/puppetlabs/bolt/blob/main/lib/bolt/fiber_executor.rb#L36L38 

It is a bit unclear to me if these are effectively the same, or if they both kind of do the same thing. I'm not quite sure how to get nested scopes meaningfully. Why were we previously trying to recurse through "parent" scopes if we only want the current scope? Also, does `to_hash` already do that recursion?

Reply:
I can at least tell you why it's like that - I don't think it's wrong, but even if it is at least it'll be wrong in the same way as the fiber executor itself.

Before the fiber executor refactor that was how the executor would collect variables https://github.com/puppetlabs/bolt/blob/66d2908c4e0b9d7b0833e1823d9dd0d064859c59/lib/bolt/executor.rb#L385 (I looked through my notes, I remember there being an issue with parallel MatchScopes being loaded into blocks which would make variables from other futures available in all futures, but I can't reproduce it now). I think we realized they were equivalent and updated the fiber executor with the refactor https://github.com/puppetlabs/bolt/pull/2810/files#diff-893ebf9461c76f97dee50a194ee4d6a3529c2383a8cac903c83a2a883312837eR37 but just didn't update BoltSpec.